keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Argmax-based box matching"
+
+from typing import List
+from typing import Tuple
+
+import tensorflow as tf
+
+
+class ArgmaxBoxMatcher:
+    """Box matching logic based on argmax of highest value (e.g., iou).
+
+    This class computes matches from a similarity matrix. Each row will be
```

Comment:
what's the row and column here? Is each row a ground truth box? Does it mean the user input (IoU matrix) need to follow this format?

Reply:
responding to the questions below as well: row is always anchors, and col is always gt_boxes. We could evaluate swapping these two, but since this is the existing implementation for many models, I'd like to get this tested within KerasCV first, and then try to swap them and evaluate the performance difference.